xc/xc7/fasm2bels/verilog_modeling.py

```diff
@@ -1639,14 +1632,12 @@ def output_nets(self):
                 assert net_wire_pkey in [ZERO_NET, ONE_NET]
                 continue
 
-            yield """
-set route_with_dummy {fixed_route}
-""".format(fixed_route=' '.join(fixed_route))
+            yield """set route {fixed_route}""".format(
+                fixed_route=' '.join(fixed_route)
+            )
 
             # Remove extra {} elements required to construct 1-length lists.
-            yield """\
-regsub -all {{}} $route_with_dummy "" route
-set_property FIXED_ROUTE $route $net"""
+            yield """set_property FIXED_ROUTE $route $net"""
```

Comment:
This works in xdc parser?  That's really great!

Reply:
Indeed, what happens though is the following, from what I understood:
- the $route and $net variables are being evaluated (they include some tcl getters such as get_node and such)
- the evaluation result of those variables is substituted into the set_property command
- the XDC is being overwritten with a "standard" XDC which is the resulting one after the parsing pass.

In other words:
```
set pin [get_pins *CLBLM_R_X25Y92_SLICE_X38Y92_RAM64X1D_CD/DPO]
set net [get_nets -of_object $pin]
set route [list  [get_nodes -of_object [get_wires CLBLM_R_X25Y92/CLBLM_M_C]] CLBLM_R_X25Y92/CLBLM_M_CMUX CLBLM_R_X25Y92/CLBLM_LOGIC_OUTS22 INT_R_X25Y92/NL1BEG_N3 INT_R_X25Y92/FAN_ALT1 INT_R_X25Y92/FAN_BOUNCE1 INT_R_X25Y92/BYP_ALT4 INT_R_X25Y92/BYP4 CLBLM_R_X25Y92/CLBLM_M_BX  ]
set_property FIXED_ROUTE $route $net
```

The above original XDC (that comes out of fasm2bels) gets overwritten wth this:
```
set_property FIXED_ROUTE { { CLBLM_M_C CLBLM_M_CMUX CLBLM_LOGIC_OUTS22 NL1BEG_N3 FAN_ALT1 FAN_BOUNCE1 BYP_ALT4 BYP4 CLBLM_M_BX }  } [get_nets CLBLM_R_X25Y95_SLICE_X38Y95_RAM64X1D_CD/DPO]
```

Reply:
Okay, your example shows why using `{}` all the time won't work, and it shows a weakness in this new approach.  I don't have a solution, but I can talk through what is happening.

```
set route [list  [get_nodes -of_object [get_wires CLBLM_R_X25Y92/CLBLM_M_C]] CLBLM_R_X25Y92/CLBLM_M_CMUX CLBLM_R_X25Y92/CLBLM_LOGIC_OUTS22 INT_R_X25Y92/NL1BEG_N3 INT_R_X25Y92/FAN_ALT1 INT_R_X25Y92/FAN_BOUNCE1 INT_R_X25Y92/BYP_ALT4 INT_R_X25Y92/BYP4 CLBLM_R_X25Y92/CLBLM_M_BX  ]
```

In the example above, `[get_nodes -of_object [get_wires CLBLM_R_X25Y92/CLBLM_M_C]]` needs to be evaluated to become `CLBLM_R_X25Y92/CLBLM_M_C`, which Vivado reduces to `CLBLM_M_C` in the output XDC.   In TCL when you use `{}` it supresses `[]` expansion.

If you run the following commands in TCL you can see this in action.

Run:
```
set l1 [list [get_nodes -of_object [get_wires CLBLM_R_X25Y92/CLBLM_M_C]] CLBLM_R_X25Y92/CLBLM_M_CMUX]
set l2 { [get_nodes -of_object [get_wires CLBLM_R_X25Y92/CLBLM_M_C]] CLBLM_R_X25Y92/CLBLM_M_CMUX }
set l3 { CLBLM_R_X25Y92/CLBLM_M_C CLBLM_R_X25Y92/CLBLM_M_CMUX }
puts [expr $l1 == $l2]
puts [expr $l1 == $l3]
```